tests/net/deckard/client.rs

```diff
@@ -67,8 +81,21 @@ fn entry2reqmsg(entry: &Entry) -> RequestMessage<Vec<u8>> {
     if reply.rd {
         msg.header_mut().set_rd(true);
     }
+    if reply.ad {
```

Comment:
Maybe this whole block could be simplified by replacing the pattern:

```
if reply.ad {
    msg.header_mut().set_ad(true);
}
```

With:

```
let mut header = msg.header_mut();
header.set_rd(reply.rd);
header.set_ad(reply.ad);
header.set_cd(reply.cd);
```

And so on...